lib/index.js

```diff
@@ -1,32 +1,113 @@
-var Logger = require('./logger');
-var transportFactory = require('./transport-factory');
-var os = require('os');
-var RollbarHandler = require('./rollbar-handler');
-var merge = require('merge');
-
-exports.createLogger = function (opts) {
-  // clone opts
-  opts = merge(true,opts);
-
-  if (!opts.system) opts.system = os.hostname();
-  if (!opts.service) opts.service = 'root';
-  if (!opts.exceptionHandlers) opts.exceptionHandlers = [
-    printStackTrace
-  ];
-  if (!opts.requestHandlers) opts.requestHandlers = [];
-
-  if (opts.rollbar && opts.rollbar.enabled) {
-    var rollbarHandler = new RollbarHandler(opts.rollbar);
-    opts.exceptionHandlers.push(rollbarHandler.logError.bind(rollbarHandler));
-    opts.requestHandlers.push(rollbarHandler.logRequest.bind(rollbarHandler));
+'use strict'
+
+var LEVELS = ['debug', 'info', 'notice', 'warn', 'error']
+
+var Syslog = require('winston-syslog').Syslog
+
+var hostname = require('os').hostname
+var winston = require('winston')
+var chalk = require('chalk')
+chalk.enabled = true
+
+var defaultColor = function (str) { return str }
+
+var colors = {
+  debug: defaultColor,
+  info: defaultColor,
+  notice: chalk.yellow,
+  warn: chalk.yellow,
+  error: chalk.red
+}
+
+function methodForLevel (method) {
+  // Syslog uses 'warning' instead of 'warn'
+  if (method === 'warn') return 'warning'
+  return method
+}
+
+module.exports = LOKELogger
+
+module.exports.create = LOKELogger
+
+function LOKELogger (options) {
+  var winstonLogger
+  var consoleLogger
+
+  function write (component, level, message) {
+    var method = methodForLevel(level)
+    var color = colors[level]
+
+    if (consoleLogger) {
+      // Write to the console:
+      consoleLogger.error( // eslint-disable-line no-console
+        chalk.yellow('«' + component + '» ') + colors[level](message)
+      )
+    }
+
+    if (winstonLogger) {
+      // Write to syslog:
+      winstonLogger.log(
+        method,
+        chalk.yellow(component) + ': ' + color(message)
+      )
+    }
   }
 
-  var transports = transportFactory(opts);
-  return new Logger(transports, opts);
-};
+  function Logger (prefix) {
+    this.component = prefix
+  }
+
+  LEVELS
```

Comment:
Yeah, only the sub loggers do. The reason to only allow sub loggers and not a root logger is for consistency.